src/datepicker/datepicker-input.ts

```diff
@@ -221,6 +221,9 @@ export class NgbInputDatepicker implements OnChanges,
         this.close();
       });
 
+      // forward disabled state
+      this._cRef.instance.setDisabledState(this._elRef.nativeElement.disabled);
```

Comment:
This would mean reading state from the DOM and we don't want to do it. We need to read it from a form control or introduce private state.